connectors/sources/gmail.py

```diff
@@ -0,0 +1,354 @@
+
+    name = "GMail"
+    service_type = "gmail"
+    advanced_rules_enabled = True
+    dls_enabled = True
+
+    def __init__(self, configuration):
+        super().__init__(configuration=configuration)
+
+    @classmethod
+    def get_default_configuration(cls):
+        """Get the default configuration for the GMail connector.
+        Returns:
+            dict: Default configuration.
+        """
+        default_credentials = {
+            "type": "service_account",
+            "project_id": "dummy_project_id",
+            "private_key_id": "abc",
+            "private_key": "",
+            "client_email": "123-abc@developer.gserviceaccount.com",
+            "client_id": "123-abc.apps.googleusercontent.com",
+            "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+            "token_uri": "http://localhost:4444/token",
+        }
+
+        return {
+            "service_account_credentials": {
+                "display": "textarea",
+                "label": "GMail service account JSON",
+                "order": 1,
+                "type": "str",
+                "value": json.dumps(default_credentials),
+                "required": True,
+            },
+            "subject": {
+                "display": "text",
+                "label": "Subject",
+                "order": 2,
+                "tooltip": "Admin account email address",
+                "type": "str",
+                "value": "subject",
+                "required": True,
+            },
+            "customer_id": {
+                "display": "text",
+                "label": "Google customer id",
+                "order": 3,
+                "tooltip": "Google admin console -> Account -> Settings -> Customer Id",
+                "type": "str",
+                "value": "",
+                "required": True,
+            },
+            "use_document_level_security": {
+                "display": "toggle",
+                "label": "Enable document level security",
+                "order": 4,
+                "tooltip": "Document level security ensures identities and permissions set in GMail are maintained in Elasticsearch. This enables you to restrict and personalize read-access users have to documents in this index. Access control syncs ensure this metadata is kept up to date in your Elasticsearch documents.",
+                "type": "bool",
+                "value": True,
+            },
+        }
+
+    async def validate_config(self):
+        """Validates whether user inputs are valid or not for configuration field.
+        Raises:
+            Exception: The format of service account json is invalid.
+        """
+        self.configuration.check_valid()
+
+        try:
+            json.loads(self.configuration["service_account_credentials"])
+        except ValueError as e:
+            raise ConfigurableFieldValueError(
+                f"Google Drive service account is not a valid JSON. Exception: {e}"
+            ) from e
```

Comment:
This validation exists in GDrive connector and is different, logic ideally should be reused, if expected fields in config are the same.

Reply:
Tracked by [Consolidate GDrive and GMail connector](https://github.com/elastic/enterprise-search-team/issues/5593)